### ISSUE TITLE
Flow.onCompletion: Documentation does not match behaviour

### ISSUE THREAD

**Issue by Mahoney** [Kotlin/kotlinx.coroutines#2009]
The documentation here:
https://kotlinlang.org/docs/reference/coroutines/flow.html#successful-completion

claims that executing the example will print:
```
1
Flow completed with java.lang.IllegalStateException: Collected 2
Exception in thread "main" java.lang.IllegalStateException: Collected 2
```

However, it does not; it prints:
```
1
Flow completed with null
Exception in thread "main" java.lang.IllegalStateException: Collected 2
  <stacktrace>
```

**Comment by Mahoney** [comment]
It works as advertised locally using 1.3.6, and if I checkout version-1.3.6 the build fails if I alter flow.md to not match FlowGuideTest.kt, or alter both to expect null, so I presume it's an issue with an older version of kotlinx.coroutines actually being run by whatever does the online execution of the examples on https://kotlinlang.org/docs/reference/coroutines.

**Comment by Mahoney** [comment]
The kotlin playground is still using 1.3.5 for running examples on the 1.3.72 branch:
https://github.com/AlexanderPrendota/kotlin-compiler-server/blob/1.3.72/build.gradle.kts#L59

It has been updated to 1.3.6 on master:
https://github.com/AlexanderPrendota/kotlin-compiler-server/blob/master/build.gradle.kts#L59

Is there any way that you can specify which version of `kotlinx-coroutines` to use when compiling & running using kotlin playground?

**Comment by Mahoney** [comment]
(Incidentally, great job on the executable documentation tested as part of the build! Will be stealing that for my next project... just needs a way to dynamically specify dependency versions.)

**Comment by qwwdfsad** [comment]
Great investigation!
This is indeed the problem of a previous version combined with a test setup that is fixed in 1.3.6.
Unfortunately, there is no way to specify coroutines version, so let's wait for https://youtrack.jetbrains.com/issue/KT-38826

>great job on the executable documentation tested as part of the build!

This is done by[ Knit tool](https://github.com/Kotlin/kotlinx-knit), check it out!
